filters/private/Point.cpp

```diff
@@ -44,74 +44,92 @@ namespace
 {
 
 const double LOWEST = (std::numeric_limits<double>::lowest)();
-const double HIGHEST = (std::numeric_limits<double>::max)();
 
 }
 
 Point::Point()
-    : Geometry()
-    , x(LOWEST)
-    , y(LOWEST)
-    , z(LOWEST)
-{}
+{
+    m_geom.reset(new OGRPoint());
+    clear();
+}
 
 
 Point::Point(const std::string& wkt_or_json, SpatialReference ref)
-    : Geometry(wkt_or_json, ref), x(LOWEST), y(LOWEST), z(LOWEST)
+    : Geometry(wkt_or_json, ref)
 {}
 
 
 void Point::update(const std::string& wkt_or_json)
 {
     Geometry::update(wkt_or_json);
 
-    int t = GEOSGeomTypeId_r(m_geoserr.ctx(), m_geom.get());
-    if (t == -1)
-        throw pdal_error("Unable to fetch geometry point type");
-    if (t > 0)
-        throw pdal_error("Geometry type is not point!");
+    OGRwkbGeometryType t = m_geom->getGeometryType();
+    if (t != wkbPoint && t != wkbPoint25D)
+    {
+        m_geom.reset(new OGRPoint());
+        throw pdal_error("Can't set Point from string.  String doesn't "
+            "represent a string.");
```

Comment:
string doesn't represent a string? Do you mean it doesn't represent a geometry?

Reply:
Ummm, that doesn't seem a helpful message.

Reply:
I'm not sure what you're looking at.  That's not what the current code says and I haven't changed anything.  Weird.  ???